jkamz/ireporter
# #164516523 Enable social login via facebook.

#### What does this PR do?
Enable social login via facebook.
#### Description of Task to be completed?
Add backend implementation of facebook login, to allow a user to login using a facebook account.
#### How should this be manually tested?
Ensure you have the below variables in your environment.
    - FACEBOOK_APP_ID=<facebook_app_id>
    - FACEBOOK_SECRET_KEY=<facebook_secret_key>
1. Clone this repo.
   `git clone https://github.com/jkamz/ireporter.git`
2. Fetch all origin to get all branches
   `git fetch origin`
3. Checkout to `ft-facebook-social-login-164516523` branch.
   `git checkout ft-facebook-social-login-164516523`
4. Activate `virtualenv`
    `$ virtualenv -p python .venv`
    `$ source .venv/bin/activate`
5. Install requirements
   `$ pip install -r requirements.txt`
6. Start server
   `$ python api/manage.py runserver`
7. Test the endpoint on postman in the following steps:
      - `POST /api/auth/oauth/login/`
      - Enter provider as `facebook` and facebook's user access token.
      - Use the generated token to authorize and create a redflag record
#### What are the relevant pivotal tracker stories?
[#164516523](https://www.pivotaltracker.com/story/show/164516523)
#### Screenshot
![image](https://user-images.githubusercontent.com/17095461/54758139-ab57a380-4bfc-11e9-8a29-a843bd61516b.png)
